src/client/components/stage-details/stage-details.component.js

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import './stage-details.style.css';
+import SectionHeader from '../section-header/section-header.component';
+import Button from '../button/button.component';
+
+function capitalizeString(str) {
+  return str[0].toUpperCase().concat(str.slice(1, str.length));
+}
+
+export default function StageDetails({
+  cropID,
+  parametersAndLabels,
+  stages,
+  levels,
+  data,
+}) {
+  // Get the parameter data for crop with id of cropID
+  const [cropData, setCropData] = useState([]);
+
+  useEffect(() => {
+    setCropData(
+      data.filter((dataPoint) => parseInt(dataPoint.fk_crop_id, 10) === cropID),
+    );
+  }, [cropID, data]);
+
+  // Select the last stage to display as default
+  const [selectedStageNumber, setSelectedStageNumber] = useState(
+    Math.max(...stages.map((s) => s.id)),
+  );
+
+  // Get data for the currently selected stage
+  const [dataForStage, setDataForStage] = useState([]);
+
+  useEffect(() => {
+    setDataForStage(
+      cropData
+        ? cropData.filter(
+            (dataPoint) =>
+              parseInt(dataPoint.fk_crop_stage_id, 10) === selectedStageNumber,
+          )
+        : [],
+    );
+  }, [selectedStageNumber, cropData]);
```

Comment:
```javascript
useEffect(() => {
    setDataForStage(
      cropData
        ? cropData.filter(
            (dataPoint) =>
              parseInt(dataPoint.fk_crop_stage_id, 10) === selectedStageNumber,
          )
        : [],
    );
  }, [selectedStageNumber, cropData]);
```

So `cropData` should not be able to be anything else but an array the way I see it so don't see a reason to have the extract check. and if you need it, you should rather check it using `&&` and then not update it I would say.

Also this is a bit easier to read, with a variable explaining the filter 👍 

```javascript
useEffect(() => {
   const selectedStages = cropData.filter(({ fk_crop_stage_id }) => parseInt(fk_crop_stage_id, 10) === selectedStageNumber)
    setDataForStage(selectedStages);
  }, [selectedStageNumber, cropData]);
```

Reply:
Nice, thank you! ✅